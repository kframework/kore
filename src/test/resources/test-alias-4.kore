[]

module TEST-ALIAS-4

  alias f{S}() : S []

endmodule []
