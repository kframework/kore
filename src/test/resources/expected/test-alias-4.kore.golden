Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "TEST-ALIAS-4"
            , moduleSentences =
                [ ObjectSentence (SentenceAliasSentence SentenceAlias
                    { sentenceAliasAlias =
                        Alias
                            { aliasConstructor = Id Object "f"
                            , aliasParams =
                                [ SortVariable (Id Object "S")
                                ]
                            }
                    , sentenceAliasSorts = []
                    , sentenceAliasReturnSort =
                        SortVariableSort (SortVariable (Id Object "S"))
                    , sentenceAliasAttributes = Attributes []
                    })
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }